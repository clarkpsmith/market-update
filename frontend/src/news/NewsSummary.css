.NewsSummary {
  background-color: black;
}
.NewsSummary-loading {
  background-color: black;
  height: 100vh;
}

.NewsSummary {
  text-align: left;
  color: white;
}

.NewsSummary-title {
  padding: 10px;
  text-align: center;
  color: rgb(200, 210, 236);
}
