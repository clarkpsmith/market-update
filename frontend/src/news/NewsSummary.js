import React, { useEffect, useState } from "react";
import YahooFinanceApi from "../api/YahooFinanceApi";
import GeneralNewsArticle from "../mktSummary/GeneralNewsArticle";
import "./NewsSummary.css";
import ChaseLoading from "../chaseloading/ChaseLoading";

const NewsSummary = ({ numberOfArticles }) => {
  const [loading, setLoading] = useState(true);
  const [newsData, setNewsData] = useState(null);

  useEffect(() => {
    async function getSummary() {
      const res2 = await YahooFinanceApi.getStockNewsSummary();
      let newsArray = [];
      for (let i = 0; i < numberOfArticles; i++) {
        newsArray.push(
          <GeneralNewsArticle key={res2[i].link} data={res2[i]} />
        );
      }

      setNewsData(newsArray);
      setLoading(false);
    }
    getSummary();
  }, [numberOfArticles]);

  if (loading)
    return (
      <div className="NewsSummary-loading">
        <ChaseLoading />
      </div>
    );

  return (
    <div className="NewsSummary">
      <h3 className="NewsSummary-title">Market News</h3>
      {newsData}
      <br />
    </div>
  );
};

export default NewsSummary;
