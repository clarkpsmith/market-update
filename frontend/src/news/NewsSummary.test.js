import React from "react";
import { render, waitForElementToBeRemoved } from "@testing-library/react";
import NewsSummary from "./NewsSummary";

it("renders without crashing", () => {
  render(<NewsSummary numberOfArticles={10} />);
});

it("matches snapshot", async () => {
  const { asFragment, queryByTestId } = render(
    <NewsSummary numberOfArticles={10} />
  );
  await waitForElementToBeRemoved(() => queryByTestId("loading"));

  expect(asFragment()).toMatchSnapshot();
});
