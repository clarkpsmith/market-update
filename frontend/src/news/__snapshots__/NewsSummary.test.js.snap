// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewsSummary"
  >
    <h3
      class="NewsSummary-title"
    >
      Market News
    </h3>
    <div
      class="Article"
    >
      <a
        aria-label="https://www.marketwatch.com/story/trade-desk-stock-tumbles-even-after-10-for-1-stock-split-profit-and-revenue-that-beat-expectations-2021-05-10?siteid=yhoof2&yptr=yahoo"
        class="Article-link"
        href="https://www.marketwatch.com/story/trade-desk-stock-tumbles-even-after-10-for-1-stock-split-profit-and-revenue-that-beat-expectations-2021-05-10?siteid=yhoof2&yptr=yahoo"
      />
      <h5
        class="Article-title"
      >
        Trade Desk stock tumbles even after 10-for-1 stock split, profit and revenue that beat expectations
      </h5>
      <div
        class="Article-source"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="https://www.marketwatch.com/story/bidens-plan-to-fully-tax-capital-gains-is-good-policy-11620658190?siteid=yhoof2&yptr=yahoo"
        class="Article-link"
        href="https://www.marketwatch.com/story/bidens-plan-to-fully-tax-capital-gains-is-good-policy-11620658190?siteid=yhoof2&yptr=yahoo"
      />
      <h5
        class="Article-title"
      >
        Biden’s plan to fully tax capital gains is good policy
      </h5>
      <div
        class="Article-source"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="https://finance.yahoo.com/news/buy-2-stocks-jump-over-132944441.html"
        class="Article-link"
        href="https://finance.yahoo.com/news/buy-2-stocks-jump-over-132944441.html"
      />
      <h5
        class="Article-title"
      >
        Buy These 2 New Stocks Before They Jump Over 80%, Says JPMorgan
      </h5>
      <div
        class="Article-source"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="https://www.investors.com/news/technology/palantir-stock-palantir-earnings-pltr-q1-2021/?src=A00220&yptr=yahoo"
        class="Article-link"
        href="https://www.investors.com/news/technology/palantir-stock-palantir-earnings-pltr-q1-2021/?src=A00220&yptr=yahoo"
      />
      <h5
        class="Article-title"
      >
        Palantir Stock To Report Earnings As Software Growth Stocks Struggle
      </h5>
      <div
        class="Article-source"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="https://www.barrons.com/articles/ford-stock-gets-bump-from-electric-f-150-unveiling-announcement-51620668315?siteid=yhoof2&yptr=yahoo"
        class="Article-link"
        href="https://www.barrons.com/articles/ford-stock-gets-bump-from-electric-f-150-unveiling-announcement-51620668315?siteid=yhoof2&yptr=yahoo"
      />
      <h5
        class="Article-title"
      >
        Ford Stock Gets Bump From Electric F-150 Unveiling Announcement
      </h5>
      <div
        class="Article-source"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="https://www.investors.com/news/spce-stock-virgin-galactic-earnings-q1-2021/?src=A00220&yptr=yahoo"
        class="Article-link"
        href="https://www.investors.com/news/spce-stock-virgin-galactic-earnings-q1-2021/?src=A00220&yptr=yahoo"
      />
      <h5
        class="Article-title"
      >
        Virgin Galactic To Report As Rival Eyes Start Of Space Tourism Flights
      </h5>
      <div
        class="Article-source"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="https://finance.yahoo.com/news/little-too-young-medicare-5-200000871.html"
        class="Article-link"
        href="https://finance.yahoo.com/news/little-too-young-medicare-5-200000871.html"
      />
      <h5
        class="Article-title"
      >
        A little too young for Medicare? Here are 5 other health insurance options
      </h5>
      <div
        class="Article-source"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="https://finance.yahoo.com/news/jessica-alba-honest-co-slumps-162626816.html"
        class="Article-link"
        href="https://finance.yahoo.com/news/jessica-alba-honest-co-slumps-162626816.html"
      />
      <h5
        class="Article-title"
      >
        Jessica Alba’s Honest Co. Slumps Less Than a Week After Its IPO
      </h5>
      <div
        class="Article-source"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="https://finance.yahoo.com/news/california-man-fraudulently-obtained-5m-225619351.html"
        class="Article-link"
        href="https://finance.yahoo.com/news/california-man-fraudulently-obtained-5m-225619351.html"
      />
      <h5
        class="Article-title"
      >
        California man fraudulently got $5M in COVID-19 relief money to buy a Ferrari, Bentley and Lamborghini, authorities say
      </h5>
      <div
        class="Article-source"
      />
    </div>
    <div
      class="Article"
    >
      <a
        aria-label="https://www.investors.com/research/what-is-a-dividend-branch/?src=A00220&yptr=yahoo"
        class="Article-link"
        href="https://www.investors.com/research/what-is-a-dividend-branch/?src=A00220&yptr=yahoo"
      />
      <h5
        class="Article-title"
      >
        What Is A Dividend? Plus The 5 Best Dividend Stocks Now
      </h5>
      <div
        class="Article-source"
      />
    </div>
    <br />
  </div>
</DocumentFragment>
`;
